src/Services/Cda/CdaValidateDocuments.php

```diff
@@ -30,14 +45,54 @@ public function __construct()
      */
     public function validateDocument($document, $type)
     {
+        // always validate schema XSD
         $xsd = $this->validateXmlXsd($document, $type);
-        $schema_results = $this->validateSchematron($document, $type);
+        if ($this->externalValidatorEnabled) {
+            $schema_results = $this->ettValidateCcda($document);
+        } else {
+            $schema_results = $this->validateSchematron($document, $type);
+        }
+
         $totals = array_merge($xsd, $schema_results);
 
         return $totals;
     }
 
     /**
+     * @param $xml
+     * @return array|mixed
+     */
+    public function ettValidateCcda($xml)
+    {
+        try {
+            $result = $this->ettValidateDocumentRequest($xml);
+        } catch (Exception $e) {
+            $e = $e->getMessage();
+            error_log($e);
```

Comment:
I would change this to (new SystemLogger())->errorLogCaller($e->getMessage(), ["trace" => $e->getTraceAsString()]);  This will escape any user data that could have been passed back from the exception since you are doing a web request here.